### PR TITLE
Test docs building in CI

### LINKED ISSUE #961: Test docs building in CI

### PR DESCRIPTION
**Pull request by drknzz**
Closes #961

## Introduced changes

- Add docs build job to CI workflow

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by piotmag769 — APPROVED** [review]
Looks good, can we somehow cache that? @THenry14

**Comment by THenry14** [comment]
> Looks good, can we somehow cache that? @THenry14

Potentially the mdbook install could be cached (~3m atm), the check would then last significantly less time. There is a rust caching done in this file which I suppose could work. 

let's remember to change repo settings after this is merged, to mark the check as required. I'd also suggest adding a `name` label for the check, not a fan of displaying `build-docs` but it's just a prefference and nothing I'm particularly strong about.

**Comment by drknzz** [comment]
> > Looks good, can we somehow cache that? @THenry14
> 
> Potentially the mdbook install could be cached (~3m atm), the check would then last significantly less time. There is a rust caching done in this file which I suppose could work.
> 
> let's remember to change repo settings after this is merged, to mark the check as required. I'd also suggest adding a `name` label for the check, not a fan of displaying `build-docs` but it's just a prefference and nothing I'm particularly strong about.

Added the name label and rust-cache 👍

File: .github/workflows/ci.yml
Comment (THenry14) [inline .github/workflows/ci.yml:162]: this is a nitpick, but I think we should stick to hashes when it goes for non-official actions like this one; tag can be easily overwritten by a maintainer and injected with mallicious code. I understand that someone already did the same in this file (I mean marked this action with a tag), so I won't block the review, but this is something that in my opinion should be changed (again - not necessarily in this PR).

**Review by THenry14 — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (drknzz) [reply .github/workflows/ci.yml:162]: Noticed it as well, but thought maybe there was already some discussion about it previously, since it's used in a few places in the file - will update to point to a commit 👍

**Pull request merged**
